tests/test_pivot.py

```diff
@@ -174,25 +173,63 @@ def test_pivot_longer_select(self):
         assert_frame_equal(long, expected)
 
 
-    # def test_pivot_longer_matches(self):
-    #     """ Test if pivot_longer is compatible with gr.matches as columns input
-    #     """
-    #     ### Not working yet, needs to be implmented
-    #     stang = data.df_stang_wide
-    #     long = gr.tran_pivot_longer(
-    #         stang,
-    #         columns=gr.matches("\\d+"),
-    #         names_to="var",
-    #         values_to="val"
-    #     )
-    #     expected = gr.tran_pivot_longer(
-    #         stang,
-    #         columns=["E_00","mu_00","E_45","mu_45","E_90","mu_90"],
-    #         names_to="var",
-    #         values_to="val"
-    #     )
-    #
-    #     #assert_frame_equal(long, expected)
+    def test_pivot_longer_matches(self):
+        """ Test if pivot_longer is compatible with gr.matches as columns input
+        """
+        stang = data.df_stang_wide
+        long = gr.tran_pivot_longer(
+            stang,
+            columns = gr.matches("\\d+"),
+            names_to="var",
+            values_to="val"
+        )
+
+        expected = gr.tran_pivot_longer(
+            stang,
+            columns=["E_00","mu_00","E_45","mu_45","E_90","mu_90"],
+            names_to="var",
+            values_to="val"
+        )
+
+        assert_frame_equal(long, expected)
+
+
+    def test_pivot_longer_names_pattern(self):
+        """ Test if pivot_longer properly works with names_pattern
+        """
+        stang = data.df_stang_wide
+        long = gr.tran_pivot_longer(
+            stang,
+            names_pattern="(E_|mu_)(\\d+)",
```

Comment:
A good test to implement, but the pattern should be `names_pattern="(E|mu)_(\\d+)"`.